### PR TITLE
added `optionsKnob` to addon-knobs

### PR DESCRIPTION
**Pull request by astrotim**
## What I did

Added `optionsKnob`* to addon-knobs which composes SelectType, RadiosType and BooleanType under the hood. This PR is for issue #3972 

This knobs has six display types:

- select
- multi-select
- radio
- inline-radio
- checkbox
- inline-checkbox

`radio` and `inline-radio` re-use the existing RadiosType.

`checkbox` and `inline-checkbox` introduces a new `CheckboxesType`.

`select` and `multi-select` use the [react-select](https://github.com/JedWatson/react-select) package.

*using the name `options` was causing lint errors due to options variable already existing in namespace, so I used the name `optionsKnob` to workaround. I'm open to suggestions for how to optimize this.

See: https://deploy-preview-4723--storybooks-official.netlify.com/?selectedKind=Addons%7CKnobs.withKnobs&selectedStory=optionsKnob&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs

screenshot

## How to test

Is this testable with Jest or Chromatic screenshots?

Yes, with Jest. I have added `__tests__/Options.js` with unit tests.

Does this need a new example in the kitchen sink apps?

Yes. I have added the new story at Knobs/withKnobs/optionsKnob which demonstrates the six display variants.

**Note** I have noticed a UI issue with the `select` and `multi-select` variants used in the dark theme. It looks like I need to contextually pass a text color style to the React Select component. If anyone can help me do this, that would be greatly appreciated.

Does this need an update to the documentation?

Yes. I have updated addons/knobs/README.md

If your answer is yes to any of these, please make sure to include it in your PR.

For maintainers only: Please tag your pull request with at least one of the following:
`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

**Comment by igor-dv** [comment]
When I check the preview, I see this:
image

Why should we have the old and the new knobs together for all these "multi value" things ? Can we alias the old knobs to point to this new "optionsKnobs" ?

**Comment by astrotim** [comment]
I initially intended to reuse `SelectType` but I decided to reach for [React Select](https://react-select.com/) when I started working on the multi select option, as this has a really nice UI for this interaction (except for the obvious dark theme issue in the screenshot, which I intend to fix)

I suppose I'm making a case for using React Select for the existing `select` knob.

**Comment by ndelangen** [comment]
This is awesome!

I see no issues other then the theming issue already mentioned, Awesome awesome work @astrotim

**Comment by ndelangen** [comment]
@astrotim I'd love your feedback and ideas on the knobs v2 discussion that's going on on our discord! https://discord.gg/sMFvFsG

File: addons/knobs/src/components/types/Options.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import ReactSelect from 'react-select';
+import styled from '@emotion/styled';
+
+import RadiosType from './Radio';
+import CheckboxesType from './Checkboxes';
+
+const OptionsSelect = styled(ReactSelect)({
+  width: '100%',
+  maxWidth: '300px',
+  color: 'black',
```
Comment (astrotim) [inline addons/knobs/src/components/types/Options.js]: I have pushed this fix for the `ReactSelect` text color issue in the dark theme - [see updated deployment](https://deploy-preview-4723--storybooks-official.netlify.com/?selectedKind=Addons%7CKnobs.withKnobs&selectedStory=optionsKnob&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs)

**Comment by astrotim** [comment]
Builds are passing now; there is just one visual addition that needs to be accepted via Chromatic. I’m not sure if it is appropriate for me to accept.

**Comment by astrotim** [comment]
@ndelangen what are your thoughts on accepting the pending Chromatic addition?

**Review by ndelangen — APPROVED** [review]

**Pull request merged**